src/main/java/it/pagopa/pm/gateway/service/VposService.java

```diff
@@ -220,22 +220,24 @@ private boolean checkResultCode(ThreeDS2Response response, PaymentRequestEntity
                 correlationId = ((ThreeDS2Authorization) response.getThreeDS2ResponseElement()).getTransactionId();
                 break;
             case RESULT_CODE_METHOD:
+                ThreeDS2Method methodResponse = (ThreeDS2Method) response.getThreeDS2ResponseElement();
                 responseType = response.getResponseType().name();
-                vposUrl = getMethodUrl(((ThreeDS2Method) response.getThreeDS2ResponseElement()));
-                correlationId = ((ThreeDS2Method) response.getThreeDS2ResponseElement()).getThreeDSTransId();
+                responseVposUrl = getMethodUrl(methodResponse);
+                correlationId = (methodResponse.getThreeDSTransId());
```

Comment:
```suggestion
                correlationId = methodResponse.getThreeDSTransId();
```

Reply:
Addressed in [63c1344](https://github.com/pagopa/pagopa-payment-transactions-gateway/pull/189/commits/63c1344951ba906fc72c75e3aa92cd57e915c7ed)